### PR TITLE
fix: handle no password on edit

### PR DESCRIPTION
**Pull request by tomzemp**
This change updates to make the password/authToken non-required when editing an existing configuration.  (See [DHIS2-10938](https://dhis2.atlassian.net/jira/software/c/projects/DHIS2/issues/DHIS2-10938))

With this update, you should be able to add new configurations and update existing ones. If you do not click and edit the password information on an existing configuration, the payload will not contain no password information, but will still update.

Notes:

- eventually the api GET request will not return password / authToken. Jan B. said this could take a while to implement, so for now, I am making the frontend update by just nulling out the password, authToken response from api
- as discussed on ticket, we are using the same UX as users app (users click checkbox to edit password/authToken). (This is not strictly necessary because the password fields are populated with fake placeholder values, and just behave as empty until user starts editing; the PUT update now just accepts a payload without password/authToken, and these will remain blank until you edit them) 
- I've updated authToken field for clikatell configuration to be password type (had a discussion with Joe about this in context of aggregate data exchange app and it was decided to make pat field password-type).

**Before**
image

**After**
image

While there may be additional changes on https://dhis2.atlassian.net/browse/DHIS2-16695 (to update the GET request), the app changes here would also work if password and authToken were not provided in response.

[DHIS2-10938]: https://dhis2.atlassian.net/browse/DHIS2-10938?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/sms-gateway/ViewSmsGatewayEdit/ViewSmsGatewayEdit.js [hunk 1]
```diff
@@ -47,16 +47,11 @@ const getFormComponent = (gatewayType) => {
...
-            [FIELD_PASSWORD_CONFIRMATION_NAME]: gateway[FIELD_PASSWORD_NAME],
-        }
-    }
-
-    return gateway
-}
+const getInitialValues = (gateway) => ({
+    ...gateway,
+    password: null,
+    authToken: null,
```
Comment (tomzemp) [inline src/sms-gateway/ViewSmsGatewayEdit/ViewSmsGatewayEdit.js:53]: these are nulled out for the moment, though in the future the api should not return values for these properties.

File: src/sms-gateway/ViewSmsGatewayEdit/ViewSmsGatewayEdit.js [hunk 2]
```diff
@@ -185,6 +178,7 @@ export const ViewSmsGatewayEdit = () => {
                                     ? history.push('/sms-gateway')
                                     : setShowCancelDialog(true)
                             }
+                            editMode={true}
```
Comment (Mohammer5) [inline src/sms-gateway/ViewSmsGatewayEdit/ViewSmsGatewayEdit.js:181]:
```suggestion
                            editMode
```

**Review by Mohammer5 — APPROVED** [review]
LGTM

**Review by Mohammer5 — APPROVED** [review]
Approving although I have a change request, @tomzemp and I discussed it on slack already:

It's an issue if you change the key of a confidential key/value pair as that'd probably create a new key/value pair rather than overriding the existing one. The old key/value pair would still be present, but the user wouldn't be aware until editing the gateway again.

Another issue that Tom found is that we don't want to load confidential values, but a user could simply disable the confidential checkbox and reload the page. Then the value would be visible in plain text.

The easiest solution we found is to disable editing the key and confidential inputs. This way users will have to delete the key/value pair and create a new one if they want to change the key or confidential flag, forcing them to know the value.

**Review by kabaros — APPROVED** [review]

**Pull request merged**